sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -303,7 +324,11 @@ private HttpPipeline getHttpPipeline() {
         policies.add(new BlobDecryptionPolicy(keyWrapper, keyResolver, requiresEncryption));
         String applicationId = clientOptions.getApplicationId() != null ? clientOptions.getApplicationId()
             : logOptions.getApplicationId();
-        policies.add(new UserAgentPolicy(applicationId, BLOB_CLIENT_NAME, BLOB_CLIENT_VERSION, userAgentConfiguration));
+
+        // adding modified user-agent string that will contain "azstorage-clientsideencryption/" + encryption version
+        String modifiedUserAgent = modifyUserAgentString(applicationId, userAgentConfiguration);
+        policies.add(new UserAgentPolicy(modifiedUserAgent));
```

Comment:
Just want to be sure this is a "per-call" policy and not "per-retry"? We need to make sure this string isn't added multiple times per request if there are retries. If we're not sure, you can also add a check to the policy to check if the additional string is not already present before adding it.

Reply:
`UserAgentPolicy` is a per-call policy. However, our string replacement code should check that this only added once.